full-service/src/service/wallet_trait.rs

```diff
@@ -0,0 +1,276 @@
+//! A MobileCoin wallet.
+
+use crate::db::AccountID;
+use crate::json_rpc;
+use crate::json_rpc::SubmitResponse;
+use crate::service::WalletServiceError;
+use mockall::*;
+
+/// A MobileCoin wallet.
+#[automock]
+pub trait Wallet {
+    /// An overview of this wallet.
+    fn get_wallet_status(&self) -> Result<json_rpc::WalletStatus, WalletServiceError>;
+
+    /// Create a new account.
+    ///
+    /// # Arguments
+    /// * `name` - The name of the account.
+    /// * `first_block` - ???
+    fn create_account(
+        &self,
+        name: Option<String>,
+        first_block: Option<u64>,
+    ) -> Result<json_rpc::CreateAccountResponse, WalletServiceError>;
+
+    /// Import an existing account into this wallet.
+    ///
+    /// # Arguments
+    /// * `entropy` - ???
+    /// * `name` - ???
+    /// * `first_block` - ???
+    fn import_account(
+        &self,
+        entropy: String,
```

Comment:
My guess: hex encoded 32 bytes from which account keys are derived. It should probably be a &[u8] or &[u8; 32] since the hex encoding is JSON-specific.

Reply:
I'd like to leave this for another PR that focuses on getting JSON out of the service layer.